android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt

```diff
@@ -0,0 +1,199 @@
+                    // your code goes here ...
+                    onNavigateToDestination = {
+                        navController.navigate(it.route) {
+                            popUpTo(navController.graph.findStartDestination().id) {
+                                saveState = true
+                            }
+                            launchSingleTop = true
+                            restoreState = true
+                        }
+                    },
+                    currentDestination = navBackStackEntry?.destination
+                )
+        }
+    ) { padding ->
+        Surface(
+            color = MaterialTheme.colorScheme.background,
+            modifier = Modifier.fillMaxSize()
+        ) {
+            NavHost(
+                navController = navController,
+                startDestination = NavigationItem.HomeDestination.route,
+                modifier = Modifier.padding(padding)
+            ) {
+                composable(NavigationItem.HomeDestination.route) {
+                    HomeRoute(
+                        onNavigateToMovieDetails = {
+                            navController.navigate(
+                                MovieDetailsDestination.createNavigationRoute(it.id)
+                            )
+                        }
+                    )
+                }
+
+                composable(NavigationItem.FavoritesDestination.route) {
+                    FavoritesRoute(
+                        onNavigateToMovieDetails = {
+                            navController.navigate(
+                                MovieDetailsDestination.createNavigationRoute(it.id)
+                            )
+                        }
+                    )
+                }
+
+                composable(
+                    route = MovieDetailsDestination.route,
+                    arguments = listOf(navArgument(MOVIE_ID_KEY) { type = NavType.IntType }),
+                ) {
+                    MovieDetailsRoute()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun TopBar(
+    navigationIcon: @Composable (() -> Unit)? = null,
+) {
+    TopAppBar(
+        backgroundColor = MaterialTheme.colorScheme.primary,
+        contentColor = MaterialTheme.colorScheme.onPrimary,
+        modifier = Modifier
+            .fillMaxHeight(0.07f)
+    ) {
+        Box(
+            modifier = Modifier
+                .fillMaxSize(1f),
+            contentAlignment = Alignment.Center,
+        ) {
+            Image(
+                painter = painterResource(id = R.drawable.tmdb_logo),
+                contentDescription = null,
+            )
+            if (navigationIcon != null) {
+                navigationIcon()
+            }
+        }
+    }
+}
+
+@Composable
+private fun BackIcon(
+    onBackClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier
+            .fillMaxSize(1f),
+        contentAlignment = Alignment.CenterStart
```

Comment:
We don't need to wrap the icon in an additional Layout Composable.
Also, the Icon should have a fixed size.